spec/features/reg_users/cancel_order_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'rails_helper'
+
+RSpec.describe 'User cancels an order', type: :feature do
+  context 'as a registered user viewing an order show page' do
+    before :each do
+      @merchant = create(:merchant)
+      @item_1 = create(:item, user: @merchant, quantity: 10)
+      @item_2 = create(:item, user: @merchant, quantity: 10)
+
+      @user = create(:user)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user)
+      @order = create(:order, user: @user, status: 0)
+
+      @unfulfilled_order_item = create(:order_item, order: @order, item: @item_1, quantity: 1)
+      @fulffileed_order_item = create(:fulfilled_order_item, order: @order, item: @item_2, quantity: 1)
+
+      visit order_path(@order)
+    end
+
+    scenario 'if the order is pending, I see a link to cancel the order' do
+      expect(page).to have_link("Cancel Order")
+    end
+
+    scenario 'if the order is NOT pending, I do not see a link to cancel the order' do
+      @packaged_order = create(:packaged_order)
```

Comment:
I would recommend not using instance variable instantiation in individual test in case there is overlap in future test in this same describe block given the length of these test files.

Reply:
Agreed. Changed them to local variables.